### PR TITLE
Global exception handler

### PR DESCRIPTION
**Pull request by VlasenkoYevhenii**

### CODE REVIEW

**Review by MolchanovAlexander — APPROVED** [review]
Good work!

File: src/main/java/com/example/bookstoreapplication/exception/CustomGlobalExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @ExceptionHandler({DataProcessingException.class, EntityNotFoundException.class})
+    protected ResponseEntity<Object> handleCustomException(Exception ex) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put(TIMESTAMP, LocalDateTime.now());
+        body.put(STATUS, HttpStatus.BAD_REQUEST);
+        body.put(MESSAGE, ex.getMessage());
+        return new ResponseEntity<>(body, HttpStatus.BAD_REQUEST);
+    }
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstoreapplication/exception/CustomGlobalExceptionHandler.java:50]: Are you sure that on EntityNotFoundException we should throw BAD_REQUEST?

File: src/main/java/com/example/bookstoreapplication/exception/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            HttpStatusCode status,
+            WebRequest request) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put(TIMESTAMP, LocalDateTime.now());
+        body.put(STATUS, HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult()
+                .getAllErrors()
+                .stream()
+                .map(this::getErrorMessage).toList();
+        body.put(ERROR, errors);
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstoreapplication/exception/CustomGlobalExceptionHandler.java:39]: Why only error is defined here?

File: src/main/java/com/example/bookstoreapplication/controller/BookController.java [hunk 3]
```diff
@@ -21,6 +22,7 @@
 @RestController
 @RequestMapping("/api/books")
 @RequiredArgsConstructor
+@Validated
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstoreapplication/controller/BookController.java:25]: I believe you should add Valid annotation in specific method to make validation work

File: src/main/java/com/example/bookstoreapplication/dto/CreateBookRequestDto.java [hunk 4]
```diff
@@ -20,7 +20,7 @@ public class CreateBookRequestDto {
     private String isbn;
 
     @NotNull(message = "Define the book price")
-    @DecimalMin(value = "0.01", message = "The price must be greater than 0")
+    @Positive(message = "The price must be greater than 0")
     private BigDecimal price;
 
     private String description;
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstoreapplication/dto/CreateBookRequestDto.java:26]: Please add some more validation rules, for example, ISBN definitely should not be empty

File: src/main/java/com/example/bookstoreapplication/exception/CustomGlobalExceptionHandler.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+        return new ResponseEntity<>(body, headers, status);
+    }
+
+    @ExceptionHandler({DataProcessingException.class, EntityNotFoundException.class})
+    protected ResponseEntity<Object> handleCustomException(Exception ex) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put(TIMESTAMP, LocalDateTime.now());
+        body.put(STATUS, HttpStatus.BAD_REQUEST);
+        body.put(MESSAGE, ex.getMessage());
+        return new ResponseEntity<>(body, HttpStatus.BAD_REQUEST);
```
Comment (andrii-hoienko) [inline src/main/java/com/example/bookstoreapplication/exception/CustomGlobalExceptionHandler.java:49]: Are you sure that on DataProcessingException we should throw BAD_REQUEST?

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Good job, but some improvements required

File: src/main/java/com/example/bookstoreapplication/dto/CreateBookRequestDto.java (hunk 4 above)
Comment (VlasenkoYevhenii) [reply src/main/java/com/example/bookstoreapplication/dto/CreateBookRequestDto.java:26]: image
There are some validations for all the mandatory fields

File: src/main/java/com/example/bookstoreapplication/exception/CustomGlobalExceptionHandler.java (hunk 5 above)
Comment (VlasenkoYevhenii) [reply src/main/java/com/example/bookstoreapplication/exception/CustomGlobalExceptionHandler.java:49]: It was supposed for only EntityNotFoundException, but there is a better HTTP status even for this one =)

File: src/main/java/com/example/bookstoreapplication/exception/CustomGlobalExceptionHandler.java (hunk 2 above)
Comment (VlasenkoYevhenii) [reply src/main/java/com/example/bookstoreapplication/exception/CustomGlobalExceptionHandler.java:39]: image
If I get it right, not only error is defined

**Pull request merged**